src/windows/wsladiag/main.cpp

```diff
@@ -0,0 +1,153 @@
+
+int wsladiag_main(std::wstring_view commandLine)
+{
+    wslutil::ConfigureCrt();
+    wslutil::InitializeWil();
+
+    WslTraceLoggingInitialize(WslaTelemetryProvider, !wsl::shared::OfficialBuild);
+    auto cleanupTelemetry = wil::scope_exit_log(WI_DIAGNOSTICS_INFO, []() { WslTraceLoggingUninitialize(); });
+
+    wslutil::SetCrtEncoding(_O_U8TEXT);
+
+    auto coInit = wil::CoInitializeEx(COINIT_MULTITHREADED);
+    wslutil::CoInitializeSecurity();
+
+    WSADATA data{};
+    THROW_IF_WIN32_ERROR(WSAStartup(MAKEWORD(2, 2), &data));
+    auto wsaCleanup = wil::scope_exit_log(WI_DIAGNOSTICS_INFO, []() { WSACleanup(); });
+
+    // Command-line parsing using ArgumentParser.
+    ArgumentParser parser(std::wstring{commandLine}, L"wsladiag");
+
+    bool help = false;
+    bool list = false;
+
+    parser.AddArgument(list, L"--list");
+    parser.AddArgument(help, L"--help", L'h'); //  short option is a single wide char
+    parser.Parse();
+
+    auto printUsage = []() {
+        wslutil::PrintMessage(
+            L"wsladiag - WSLA diagnostics tool\n"
+            L"Usage:\n"
+            L"  wsladiag --list    List WSLA sessions\n"
+            L"  wsladiag --help    Show this help",
+            stderr);
+    };
+
+    // If '--help' was requested, print usage and exit.
+    if (help)
+    {
+        printUsage();
+        return 0;
+    }
+
+    if (!list)
+    {
+        // No recognized command → show usage
+        printUsage();
+        return 0;
+    }
+
+    // --list: Call WSLA service COM interface to retrieve and display sessions.
+    
+    try
+    {
+        wil::com_ptr<IWSLAUserSession> userSession;
+        THROW_IF_FAILED(CoCreateInstance(__uuidof(WSLAUserSession), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&userSession)));
+
+        wsl::windows::common::security::ConfigureForCOMImpersonation(userSession.get());
+
+        wil::unique_cotaskmem_array_ptr<WSLA_SESSION_INFORMATION> sessions;
+
+        THROW_IF_FAILED(userSession->ListSessions(&sessions, sessions.size_address<ULONG>()));
+
+        // Free inner CoTaskMem-allocated strings before the array is freed.
+        auto cleanupInnerStrings = wil::scope_exit([&]() {
+            for (ULONG i = 0; i < sessions.size(); ++i)
+            {
+                if (sessions[i].DisplayName != nullptr)
+                {
+                    CoTaskMemFree(sessions[i].DisplayName);
+                    sessions[i].DisplayName = nullptr;
+                }
+            }
+        });
+
+        if (sessions.size() == 0)
+        {
+            wslutil::PrintMessage(L"No WSLA sessions found.\n", stdout);
+        }
+        else
+        {
+            wslutil::PrintMessage(std::format(L"Found {} WSLA session{}:\n", sessions.size(), sessions.size() > 1 ? L"s" : L""), stdout);
+
+            wslutil::PrintMessage(L"ID\tCreator PID\tDisplay Name\n", stdout);
+            wslutil::PrintMessage(L"--\t-----------\t------------\n", stdout);
```

Comment:
This is probably OK as the first implementation, but long term (maybe in a followup PR), we'll want to compute what's the longest string that we'll display, so we can have the right number of columns displayed here so everything is alligned regardless of the DisplayName's sizes

Reply:
Here's how do it for wsl --list for reference: https://github.com/microsoft/WSL/blob/ba90ee11fa83a1cb7d908d64cb50197afa6a8aeb/src/windows/common/WslClient.cpp#L730C8-L738C12 

(We could probably do a more modern version of that)